aten/src/ATen/core/TensorBase.h

```diff
@@ -205,18 +205,8 @@ class TORCH_API TensorBase {
     impl_.reset();
   }
 
-  TensorBase& operator=(const TensorBase& x) & {
-    impl_ = x.impl_;
-    return *this;
-  };
-  TensorBase& operator=(TensorBase&& x) & noexcept {
-    impl_ = std::move(x.impl_);
-    return *this;
-  }
-
-  // Ban assignment to rvalues, since at::Tensor (weirdly) performs a deep copy here
```

Comment:
Are we sure this is no longer an issue? Do we have a test case for this?

Reply:
This is still legit. It won't be covered by tests because it is disabling a footgun

Reply:
Assignment to rvalue is actually strange. This should be an indication of logic errors.

Reply:
In Tensor, it's to handle idioms like this: `tensor[0] = blah`. The `tensor[0]` is an rvalue and the assignment lets us copy the data from blah into the slice. When @peterbell10 refactored TensorBase out, I guess he thought this behavior was weird and disabled it on TensorBase.